### DISCUSSION TITLE
Only show "updated on" on articles that have been edited

### DISCUSSION THREAD

**Discussion by Backspaze** [GetPublii/Publii#1657] [Ideas]
I'd like to see an option for only showing the last updated date on articles that have been edited after the publish date.

There are themes where the publish date and updated date are displayed close to each other, and I think it looks weird to have the last updated information there if there haven't been any changes since the publish date.

**Comment by bobmitro** [comment]
Usually, you can disable the published or updated date. Which theme are you using?

**Reply by Backspaze** [reply]
Yes, there are both global options, and per post/page options for what information should be displayed. Using those options, the closest you could get to what I'm looking for is to set the global option to not display the last updated date and instead change that on every post/page you edit.

I'd prefer if you could have the global option set to display both the publish and last updated date, but the latter would only be displayed if the post/page has been edited since the publish date. That way it's handled automatically and you wouldn't have to worry about remembering to change that setting on every post/page you edit.

I'm guessing one reason why this feature doesn't exist (yet) is that it would have to be handled during the build process, and it would have to check every page/post to see if it has been edited. If so, I suspect that it would impact the build time for large sites. But maybe there's some smart method that could be implemented that checks for what files has been edited and only checks those for adding the last updated date. Note that I have no clue how the build process works, so maybe it already does this.

**Comment by bobmitro** [comment]
The app already checks your changes, so it doesn't make much difference. I'll add this during the next themes update, as it's a good idea. But for now, if you want to implement it, you'll need to add a check to see if the modification date differs from the publication date.

`
{{#checkIf modifiedAt '!=' createdAt}}
   {{#if modifiedAt}}
     ...
   {{/if}}
{{/checkIf}}
`

**Reply by Backspaze** [reply]
I just did a quick test and it seems to be working, except perhaps on one post (see below), for all other posts it looks like it's working.

In the editor, that post is shown like this: 

![bild](https://github.com/user-attachments/assets/9c36d7fc-7b6b-4c0b-9571-2cfe3b6ac8b2)

On the preview page, it's shown like this:

![bild](https://github.com/user-attachments/assets/9bbef333-37fa-40a4-a913-0d35e4f5ce3b)

I'm guessing that the reason for this is that `checkIf modifiedAt '!=' createdAt` checks for seconds (and perhaps milliseconds) as well. So the post may have been, for example, published at 19:43:10 and updated at 19:43:50.

This is how I added the relevant code in my `post.hbs`:

```handlebars
{{#if @config.post.displayUpdatedDate}}
  {{#checkIf modifiedAt '!=' createdAt}}
    {{#if modifiedAt}}
      <div class="content__last-updated">
        {{ translate 'post.lastUpdatedDate' }}
        <time datetime="{{date modifiedAt 'YYYY-MM-DDTHH:mm'}}">
          {{#checkIf @config.custom.formatDate '!=' 'custom'}}
            {{date modifiedAt @config.custom.formatDate}}
          {{else}}
            {{date modifiedAt @config.custom.formatDateCustom}}
          {{/checkIf}}
        </time>
      </div>
    {{/if}}  
  {{/checkIf}}
{{/if}}
```
